subsys/bluetooth/host/gatt.c

```diff
@@ -1743,10 +1743,18 @@ int bt_gatt_service_register(struct bt_gatt_service *svc)
 
 int bt_gatt_service_unregister(struct bt_gatt_service *svc)
 {
+	uint16_t sc_start_handle;
+	uint16_t sc_end_handle;
 	int err;
 
 	__ASSERT(svc, "invalid parameters\n");
 
+	/* gatt_unregister() clears handles when those were auto-assigned
+	 * by host
+	 */
+	sc_start_handle = svc->attrs[0].handle;
+	sc_end_handle = svc->attrs[svc->attr_count - 1].handle;
```

Comment:
Probably outside the scope of this PR, but we probably should verify that `svc->attrs != NULL` and that `svc->attr_count > 0` before doing these lines